src/backend/starknet.cairo

```diff
@@ -295,4 +302,48 @@ namespace Internals {
 
         return _save_storage(starknet_address, storage_start + DictAccess.SIZE, storage_end);
     }
+
+    func _save_valid_jumpdests{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        starknet_address: felt, dict_start: DictAccess*, dict_end: DictAccess*
+    ) {
+        alloc_locals;
+        let dict_len = dict_end - dict_start;
+        if (dict_len == 0) {
+            return ();
+        }
+
+        let (local keys_start: felt*) = alloc();
+
+        tempvar keys_len = 0;
+        tempvar keys = keys_start;
+        tempvar dict = dict_start;
+        tempvar remaining = dict_len;
+
+        loop:
+        let keys_len = [ap - 4];
+        let keys = cast([ap - 3], felt*);
+        let dict = cast([ap - 2], DictAccess*);
+        let is_valid = dict.new_value;
+
+        if (is_valid != 0) {
+            assert [keys] = dict.key;
+            tempvar keys_len = keys_len + 1;
+            tempvar keys = keys + 1;
+            tempvar dict = dict + DictAccess.SIZE;
+        } else {
+            tempvar keys_len = keys_len;
```

Comment:
`keys_len` is not necessary, just recompute at the end with `keys - keys_start` (or remove `keys` and `assert [keys_start + keys_len] == dict.key`)